src/phase-2/scWETHv2.sol

```diff
@@ -0,0 +1,533 @@
+
+        afterDeposit(assets, shares);
+    }
+
+    function withdraw(uint256, address, address) public virtual override returns (uint256) {
+        revert PleaseUseRedeemMethod();
+    }
+
+    function redeem(uint256 shares, address receiver, address owner) public override returns (uint256 assets) {
+        if (msg.sender != owner) {
+            uint256 allowed = allowance[owner][msg.sender]; // Saves gas for limited approvals.
+
+            if (allowed != type(uint256).max) allowance[owner][msg.sender] = allowed - shares;
+        }
+
+        // Check for rounding error since we round down in previewRedeem.
+        require((assets = previewRedeem(shares)) != 0, "ZERO_ASSETS");
+
+        beforeWithdraw(assets, shares);
+
+        _burn(owner, shares);
+
+        uint256 balance = asset.balanceOf(address(this));
+
+        if (assets > balance) {
+            assets = balance;
+        }
+
+        emit Withdraw(msg.sender, receiver, owner, assets, shares);
+
+        asset.safeTransfer(receiver, assets);
+    }
+
+    /// @dev called after the flashLoan on rebalance
+    function receiveFlashLoan(address[] memory, uint256[] memory amounts, uint256[] memory, bytes memory userData)
+        external
+    {
+        if (msg.sender != address(balancerVault)) {
+            revert InvalidFlashLoanCaller();
+        }
+
+        _isFlashLoanInitiated();
+
+        // the amount flashloaned
+        uint256 flashLoanAmount = amounts[0];
+
+        // decode user data
+        (RebalanceParams memory rebalanceParams) = abi.decode(userData, (RebalanceParams));
+
+        // repay and withdraw first
+        for (uint8 i; i < rebalanceParams.repayWithdrawParams.length; i++) {
+            _repayWithdraw(rebalanceParams.repayWithdrawParams[i]);
+        }
+
+        if (rebalanceParams.wstEthToWethSwapData.length != 0) {
+            // wstEth to weth
+            wstEthToWethSwapRouter.functionDelegateCall(rebalanceParams.wstEthToWethSwapData);
+        }
+
+        if (rebalanceParams.wethToWstEthSwapData.length != 0) {
+            // weth to wstEth
+            wethToWstEthSwapRouter.functionDelegateCall(rebalanceParams.wethToWstEthSwapData);
+        }
+
+        for (uint8 i; i < rebalanceParams.supplyBorrowParams.length; i++) {
+            _supplyBorrow(rebalanceParams.supplyBorrowParams[i]);
+        }
+
+        // payback flashloan
+        asset.safeTransfer(address(balancerVault), flashLoanAmount);
+
+        _enforceFloat();
+    }
+
+    // need to be able to receive eth
+    receive() external payable {}
+
+    //////////////////// INTERNAL METHODS //////////////////////////
+
+    function _withdrawToVault(uint256 amount) internal {
+        uint256 n = protocolAdapters.length();
+        uint256 flashLoanAmount;
+        RepayWithdrawParam[] memory repayWithdrawParams = new RepayWithdrawParam[](n);
+
+        uint256 totalInvested_ = totalCollateral() - totalDebt();
+
+        {
```

Comment:
```suggestion

```

Reply:
this pair of brackets seems redundant

Reply:
Initially I added to solve the "STACK TOO DEEP ERROR" but that is not the issue now.

I would argue that it adds to the readability to the code by telling that the variables inside those brackets are temporary and not used outside of those brackets.